detection_rules/rule.py

```diff
@@ -230,7 +230,16 @@ class AlertSuppressionDuration:
 
     group_by: List[definitions.NonEmptyStr]
     duration: Optional[AlertSuppressionDuration] = field(metadata=dict(metadata=dict(min_compat="8.7")))
+    missing_fields_strategy: Optional[definitions.AlertSuppressionMissing] = field(
+        metadata=dict(metadata=dict(min_compat="8.8")))
 
+    def validate(self, data: Any):
```

Comment:
Nit. Should the type hint for `data` be `Any`? It looks like the function is depending on the presence of the `.get()` function which not all types would have.

Reply:
It could be any of the `RuleData` dataclasses, which cannot be passed as types here as I have tried. I attempted to create a custom `TypeVar` being bound to `BaseRuleData` since that is inherited by `QueryRuleData` and the subsequent rule data classes, but attempting to add as a global variable doesn't work, neither does it work in `mixins.py` because of circular imports with `rule.py`.

As a result, the best option that I can think of is to type hint it as `dataclass`. Looking at annotations of `validate_recursive`, `obj` is expected to be a `dataclass` of sorts.